src/amp-story-player/amp-story-player-impl.js

```diff
@@ -177,6 +178,61 @@ export class AmpStoryPlayer {
     this.layoutCallback();
   }
 
+  /**
+   * Adds stories to the player, both adding new anchor tags to the player,
+   * and creating or assigning iframes to those that are close to the
+   * current playing story.
+   * @param {!Array<!{href: string, title: ?string, posterImage: ?string}>} stories
+   * @public
+   */
+  add(stories) {
+    const isStoryObject = (story) =>
+      typeof story === 'object' && story !== null && story.href;
+
+    if (!Array.isArray(stories) || !stories.every(isStoryObject)) {
+      throw new Error('"stories" parameter has the wrong structure');
+    }
+
+    for (let i = 0; i < stories.length; i++) {
+      const story = stories[i];
+      const anchor = this.createStoryAnchor_(story);
+
+      this.stories_.push(anchor);
+
+      if (this.iframes_.length < MAX_IFRAMES) {
+        this.createIframeForStory_(this.stories_.length - 1);
+        continue;
+      }
+
+      // If this story is after the current one
+      if (this.stories_[this.currentIdx_ + 1] === anchor) {
+        this.allocateIframeForStory_(this.currentIdx_ + 1);
+        continue;
+      }
+    }
+  }
+
+  /**
+   * Given a story object, creates an appropiate anchor element.
+   * @param {{href: string, title: ?string, posterImage: ?string}} story
+   * @return {HTMLAnchorElement}
+   * @private
+   */
+  createStoryAnchor_(story) {
+    const anchor = this.doc_.createElement('a');
+    anchor.href = story.href;
+    story.posterImage &&
+      anchor.setAttribute('data-poster-portrait-src', story.posterImage);
+
+    const title = this.doc_.createElement('span');
```

Comment:
I think we should either:

1) validate the story has a title when consuming it in `add()`
2) check if the story has title here and only create the element if it does.

Reply:
I went with the 2nd option, so now the span element wont be created unless there is a title